src/llm_vm/onsite_llm.py

```diff
@@ -279,3 +359,30 @@ def generate(self,prompt, max_length=100,**kwargs): # both tokenizer and model t
             model="gpt-3.5-turbo-0301",
             **kwargs)
         return ans['choices'][0]['message']['content']
+
+    def finetune(self, dataset, optimizer, c_id):
+        old_model = optimizer.storage.get_model(c_id)
+        training_file = create_jsonl_file(dataset)
```

Comment:
Why are we writing the training data to disk?

Reply:
Matt's code, creates a temp jsonl file that we send to openAI. I just moved this code out from optimize and made it a memeber of the gpt class under a finetune function